tests/util/test_Executor.cpp

```diff
@@ -0,0 +1,54 @@
+#include <aikido/util/ExecutorMultiplexer.hpp>
+#include <aikido/util/ExecutorThread.hpp>
+#include <gtest/gtest.h>
+
+using namespace aikido::util;
+
+static int numCalled = 0;
+
+void foo() { numCalled++; }
+
+//==============================================================================
+TEST(ExecutorMultiplexer, Execute)
+{
+  ExecutorMultiplexer exec;
+  EXPECT_TRUE(exec.isEmpty());
+  EXPECT_TRUE(exec.getNumCallbacks() == 0u);
+
+  exec.addCallback(foo);
+  exec.addCallback([]() {});
+  EXPECT_TRUE(!exec.isEmpty());
+  EXPECT_TRUE(exec.getNumCallbacks() == 2u);
+
+  EXPECT_TRUE(numCalled == 0);
+  exec();
+  EXPECT_TRUE(numCalled == 1);
+
+  exec.removeAllCallbacks();
+  EXPECT_TRUE(exec.isEmpty());
+
+  EXPECT_TRUE(numCalled == 1);
+  exec();
+  EXPECT_TRUE(numCalled == 1);
+}
+
+//==============================================================================
+TEST(ExecutorThread, Execute)
+{
+  ExecutorThread exec([]() {}, std::chrono::nanoseconds(1));
+
+  EXPECT_TRUE(exec.isRunning());
+  exec.stop();
+  EXPECT_TRUE(!exec.isRunning());
+}
+
+//==============================================================================
+TEST(ExecutorThread, ExceptionThrownByCallback)
+{
+  ExecutorThread exec(
+      []() { throw std::exception(); }, std::chrono::milliseconds(1));
+
+  std::this_thread::sleep_for(std::chrono::seconds(3));
+
+  EXPECT_TRUE(!exec.isRunning());
```

Comment:
This is technically a race condition, but I don't see a better way of implementing it. I suppose the 3 second sleep is essentially a timeout for the test.